File: Ruby/calculator.rb
#!/usr/bin/ruby

# puts "Hello, Ruby!"

# puts "What's your name?"
# name = Aniket



# puts "\nHi #{name}"

class Calculator
  def add(a, b)
    a + b
  end
  
  def sub(a, b)
    a - b
  end

  def mul(a, b)
    a * b
  end

  def div(a, b)
    raise "Cannot divide by zero" if b.zero?
    a / b.to_f
  end
end 

calculator = Calculator.new

puts "Ruby Calculator\n"

loop do 
  puts "Please enter first number: "
  num1 = gets.chomp.to_f

  puts "Please enter second number: "
  num2 = gets.chomp.to_f

  puts "$elect an opearation(+, -, *, /): "
  operator = gets.chomp

  result = case operator
           when "+"
             calculator.add(num1, num2)
           when "-"
             calculator.sub(num1, num2)
           when "*"
             calculator.mul(num1, num2)
           when "/"
             calculator.div(num1, num2)
           else
             "Invalid operator"
           end

  puts "Result: #{result}"

  puts "Do you want to perform another calculation? (yes/no)"
  continue = gets.chomp.downcase
  break if continue != "yes"
end

File: C/calculator.c
#include <stdio.h>

int main() {
    int num1, num2;
    int sum, diff, mul, div;
    char op;

    printf("Calculator\n\n");
    printf("Please Enter first value: \n");
    scanf(" %d", &num1);
    printf("\nPlease enter operation: [+, -, *, /]\n");
    scanf(" %s", &op);
    printf("\nPlease enter second value: \n");
    scanf(" %d", &num2);
    // printf("%d,%s,%d", num1, op, num2);
 
    if (op == '+') {
        sum = num1+num2;
        printf("\nAddition of %d + %d is: %d: ", num1, num2, sum);
    }
    else if (op == '-') {
        diff = num1-num2;
        printf("Difference of %d - %d is: %d: ", num1, num2, diff);
    }
    else if (op == '*') {
        mul = num1*num2;
        printf("Multiplication of %d * %d is: %d: ", num1, num2, mul);
    }
    else if (op == '/') {
        div = num1/num2;
        printf("Divde %d / %d is: %d: ", num1, num2, div);
    }
    else {
        printf("Please enter a valid operator!");
    }
    getchar();
}